app.py

```diff
@@ -17,16 +17,38 @@
 }
 
 # Routes for add tasks 
-@app.route('/add/tasks', methods=['POST'])
+@app.route('/tasks', methods=['POST'])
 def add_task():
-    data = request.get_json()
-    task = {k: task_schema[k](v) for k, v in data.items() if k in task_schema}
-    result = mongo.db.tasks.insert_one(task)
-    return jsonify({'_id': str(result.inserted_id)})
+    try:
+        data = request.get_json()
+
+        if 'name' not in data or 'description' not in data:
+            return jsonify({'error': 'Name and description are required fields'}), 400
+        
+        if 'completed' in data and not isinstance(data['completed'], bool):
+            return jsonify({'error': 'Completed field must be a boolean'}), 400
+        
+        if not data.get('name') or not data.get('description'):
```

Comment:
This is a same condition as above, You need to check like,
if not data.get('name') == ''"

Also try it out with strip() method.